File: Day28-Reversing the equation.py
#User function Template for python3
class Solution:
    def reverseEqn( self,s):
        m,count=len(s)-1,0
        res=""
        while(m>=0): 
            if s[m] not in ["*","-","+","/"]:
                count+=1
            else:
                res+=(s[m+1:len(s)]+(s[m]))
                s=s[:m] 
                count=0 
            m-=1
        res+=s
        return res


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        s = input().strip()
        ob = Solution()
        ans = ob.reverseEqn(s)
        print(ans)
# } Driver Code Ends

File: Day28-Reversing the equation.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

//problem link 
//https://practice.geeksforgeeks.org/problems/reversing-the-equation2205/1




// } Driver Code Ends
class Solution
{
  public:
    string reverseEqn (string s)
        {
       string ans="";
            stack<string>st;
            for(int i=0;i<s.length();i++){
                if(s[i]=='+' || s[i]=='-' || s[i]=='/' || s[i]=='*'){
                    st.push(ans);
                    ans=s[i];
                    st.push(ans);
                    ans="";
                }
                else{
                ans+=s[i];
                }
            }
            st.push(ans);
            
            string res="";
            while(!st.empty()){
                res+=st.top();
                st.pop();
            }
            
            return res;
        }
};

//{ Driver Code Starts.
int main()
{
    int t; cin >> t;
    while (t--)
    {
        string s; cin >> s;
        Solution ob;
        cout <<ob.reverseEqn (s) << endl;
    }
}
// Contributed By: Pranay Bansal

// } Driver Code Ends